src/rust/engine/process_execution/docker/src/docker.rs

```diff
@@ -218,9 +218,9 @@ async fn credentials_for_image(
                 // TODO: https://github.com/keirlawson/docker_credential/issues/7 means that this will only
                 // work for credential helpers and credentials encoded directly in the docker config,
                 // rather than for general credStore implementations.
-                let credential = docker_credential::get_credential(&server).map_err(|e| {
-                    format!("Failed to retrieve credentials for server `{server}`: {e}")
-                })?;
+                let Ok(credential) = docker_credential::get_credential(&server) else {
+                    return Ok(None);
```

Comment:
Should this log a warning?

Reply:
Also, what are the values of the `Err` branch of the result? Are there any which should propagate the error?

Reply:
> Also, what are the values of the `Err` branch of the result? Are there any which should propagate the error?

The errors are relatively exhaustive (cant find the file, cant decode the file, etc...) In those cases, trying `docke rpull` out locally just warns about those things but still succeeds, so I think we ought to simply do the same.